src/IndexUpdater.php

```diff
@@ -0,0 +1,135 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\elasticsearch;
+
+use Elasticsearch\Client;
+use Elasticsearch\ClientBuilder;
+use oat\generis\model\WidgetRdf;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\search\index\IndexUpdaterInterface;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class IndexUpdater extends ConfigurableService implements IndexUpdaterInterface
+{
+    /** @var Client */
+    private $client;
+
+    protected function getClient(): Client
+    {
+        if (is_null($this->client)) {
+            $this->client = ClientBuilder::create()
+                ->setHosts($this->getOptions())
+                ->build();
+        }
+
+        return $this->client;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function updateProperties(array $properties): void
+    {
+        $queryNewProperty = [];
+        $queryRemoveOldProperty = [];
+        foreach ($properties as $propertyData) {
+            if (!isset($propertyData['type'], $propertyData['parentClasses'], $propertyData['oldName'], $propertyData['newName'])) {
+                continue;
+            }
+
+            $type = $propertyData['type'];
+            $parentClasses = $propertyData['parentClasses'];
+
+            $queryNewProperty[] = sprintf(
+                'ctx._source[\'%s\'] = ctx._source[\'%s\'];',
+                $propertyData['newName'],
+                $propertyData['oldName']
+            );
+            $queryRemoveOldProperty[] = sprintf(
+                'ctx._source.remove(\'%s\');',
+                $propertyData['oldName']
+            );
+        }
+
+        if (!isset($type, $parentClasses)) {
+            return;
+        }
+
+        $index = $this->findIndex($parentClasses, $type);
+
+        if ($index === IndexerInterface::UNCLASSIFIEDS_DOCUMENTS_INDEX) {
+            return;
+        }
+
+        $result = implode(' ', array_merge($queryNewProperty, $queryRemoveOldProperty));
+
+        try {
+            $this->getClient()->updateByQuery(
+                [
+                    'index' => $index,
+                    'type' => '_doc',
+                    'conflicts' => 'proceed',
+                    'wait_for_completion' => true,
+                    'body' => [
+                        'query' => [
+                            'match' => [
+                                'type' => $type
+                            ]
+                        ],
+                        'script' => [
+                            'source' => $result
+                        ]
+                    ]
+                ]
+            );
+        } catch (Throwable $e) {
+            throw new FailToUpdatePropertiesException(
+                sprintf(
```

Comment:
Proposal: move the first argument "processing" to the `FailToUpdatePropertiesException` itself

Reply:
I tried to encapsulate the message. without changing the exception interface (do not violate the Liskov principle). Could you please check it?